homeassistant/components/intellifire/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for IntelliFire integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from intellifire4py import IntellifireAsync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import _LOGGER, DOMAIN
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required("host"): str})
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    api = IntellifireAsync(data["host"])
+    try:
+        await api.poll()
+    except Exception:
+        raise CannotConnect
+
+    serial_number = api.data.serial
+
+    # Return info that you want to store in the config entry.
+    return {"title": "Fireplace", "type": "Fireplace", "serial": serial_number}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for IntelliFire."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+            # Store serial into 'user_input' which is going to be stored with the config entry
+            user_input["serial"] = info["serial"]
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
Set the serial as a unique ID, this prevents adding multiple instances of the same fireplace. If added again with a new address, it gets updated.

I don't think we need to store the serial, as it can be retrieved upon connecting to the device.

```suggestion
			await self.async_set_unique_id(serial)
			self._abort_if_unique_id_configured(updates={CONF_HOST: user_input[CONF_HOST]})

            return self.async_create_entry(
                title="Fireplace",
                data={
                  CONF_HOST: user_input[CONF_HOST],
                },
            )
```